contracts/mocks/docs/utilities/Base64NFT.sol

```diff
@@ -0,0 +1,27 @@
+// contracts/Base64NFT.sol
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {ERC721} from "../../../token/ERC721/ERC721.sol";
+import {Strings} from "../../../utils/Strings.sol";
+import {Base64} from "../../../utils/Base64.sol";
+
+contract Base64NFT is ERC721 {
+    using Strings for uint256;
+
+    constructor() ERC721("Base64NFT", "MTK") {}
+
+    // ...
+
+    function tokenURI(uint256 tokenId) public pure override returns (string memory) {
+        // Equivalent to:
+        // {
+        //   "name": "Base64NFT #1",
+        //   // Replace with extra ERC-721 Metadata properties
+        // }
+        bytes memory dataURI = abi.encodePacked("{", '"name": "Base64NFT #', tokenId.toString(), '"', "}");
```

Comment:
In these kind of application, I think using `string.concat` is clearer.
Also, things can be merges before.
```suggestion
        string memory dataURI = string.concat("{\"name\": \"Base64NFT #", tokenId.toString(), "\"}");
```